### PR TITLE
cy(doc): add d4rl(mujoco) English file

### PR DESCRIPTION
**Pull request by chenyingvivian**
done the translation

### CODE REVIEW

File: D4RL (Mujoco).md [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+
+-  Maze2D
+-  AntMaze
+-  Gym-MuJoco
+-  Adroit
+-  FrankaKitchen
+-  Flow
+-  Offline CARLA
+
+Note: offline RL is the dataset with d4rl for training, and tests are interacted with a specific RL environment, such as Mujoco.
```
Comment (karroyan) [inline D4RL (Mujoco).md]: "offline RL is trained on the d4rl dataset, and tested by interacting with a specific ..."

File: D4RL (Mujoco).md [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+-  expert: train a SAC algorithm online until the strategy reaches the expert performance level, using the expert strategy to collect 1 million samples of data
+-  medium-expert: mix equal amounts of data collected by expert and medium strategies
+-  medium: first train a SAC algorithm online, stop training in the middle, and then use this partially trained strategy to collect 1 million samples of data
+-  medium-replay：train a SAC algorithm online until the strategy reaches a moderate performance level, collecting all the samples placed in the buffer during training
+-  random：use a random initialization strategy to collect
+
+The picture below shows one of the Hopper games.
+
+.. image:: ./images/d4rl.gif
```
Comment (karroyan) [inline D4RL (Mujoco).md]: The picture has not shown properly

File: D4RL (Mujoco).md [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    # Each task is associated with a dataset
+    # dataset contains observations, actions, rewards, terminals, and infos
+    dataset = env.get_dataset()
+    print(dataset['observations']) # An N x dim_observation Numpy array of observations
+
+    # Alternatively, use d4rl.qlearning_dataset which
+    # also adds next_observations.
+    dataset = d4rl.qlearning_dataset(env)
+
+Mirroring
```
Comment (karroyan) [inline D4RL (Mujoco).md]: Images

File: D4RL (Mujoco).md [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+----
+
+The DI-engine is ready with images of the framework itself, which can be obtained by \ ``docker pull opendilab/ding:nightly-mujoco``, or by visiting \ `docker
```
Comment (karroyan) [inline D4RL (Mujoco).md]: 1. mirror->image
2. add space after the url

File: D4RL (Mujoco).md [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+hub <https://hub.docker.com/r/opendilab/ding>`__\ for more mirrors
+
+
+GYM-MUJOCO'S SPACE BEFORE TRANSFORMATION (ORIGINAL ENVIRONMENT)
```
Comment (karroyan) [inline D4RL (Mujoco).md]: The letters in titles do not need to be fully capitalized

File: D4RL (Mujoco).md [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+--------
+
+-  The vector consists of physical information(3D position, orientation, and joint angles etc.), and the specific size is \ ``(N, )``\, which is determined by the environment. The data type is \ ``float64``
+-  `Fujimoto <https://github.com/opendilab/DI-engine/blob/main/dizoo/d4rl/entry/d4rl_cql_main.py>`__ mentions that doing obs norm for the d4rl dataset will improve the stability of offline training.
+
+
+Action Space
+--------
+
+-  The vector consists of physical information(torque etc.), which is often the continuous action space.（N varies with specific subenvironments), the data type is \ ``float32``\ ，需要传入 np 数组（例如动作为\ ``array([-0.9266078 , -0.4958926 ,  0.46242517], dtype=float32)``\ ）
```
Comment (karroyan) [inline D4RL (Mujoco).md]: 1. "..., which is often a continuous action space with N dimension (N varies...."
2. translate the chinese here

File: D4RL (Mujoco).md [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+
+-  If it's in the Hooper, N is 3, then the action value be chosen in \ ``[-1, 1]``\.
```
Comment (karroyan) [inline D4RL (Mujoco).md]: "For example, if ..."

File: D4RL (Mujoco).md [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+
+Others
+----
+
+-  The end of the game is the end of the current environment episode
+
+KEY FACTS
+========
+
+1. Vector physical information input, empirically it is not advisable to subtract the mean value in doing norm
```
Comment (karroyan) [inline D4RL (Mujoco).md]: doing norm->normalization

File: D4RL (Mujoco).md [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+Others
+----
+
+-  The \ ``info``\ returned by the environment \ ``step``\ method must contain the \ ``final_eval_reward``\ key-value pair, which represents the evaluation metrics for the entire episode, which in Mujoco is the cumulative sum of the rewards for the entire episode.
+
+
+OTHERS
+====
+
+Inert initialization
```
Comment (karroyan) [inline D4RL (Mujoco).md]: Lazy initialization

File: D4RL (Mujoco).md [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+----------
+
+To facilitate support for parallel operations such as environment vectorization, environment instances are generally implemented with inert initialization, i.e., the \``__init__`` method does not initialize the real original environment instance, but only sets the relevant parameters and configuration values, and initializes the concrete original environment instance when the \``reset`` method is called for the first time.
```
Comment (karroyan) [inline D4RL (Mujoco).md]: 1. inert initialization -> lazy initialization
2. all the format like ``__init__`` is not shown properly

File: D4RL (Mujoco).md [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+
+
+Video storage
+--------
+
+After the environment is created and before it is reset, the \``enable_save_replay`` method is called to specify the path where the game footage is saved. The environment will automatically save the session's video files after each episode. (The default call \``gym.wrappers.RecordVideo`` is implemented), the code shown below will run an environment episode and save the results of that episode in \``. /video/`` \\.
+.. code:: python
```
Comment (karroyan) [inline D4RL (Mujoco).md]: The code format is not shown properly

File: D4RL (Mujoco).md [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+`link <https://github.com/opendilab/DI-engine/blob/main/dizoo/d4rl/entry/d4rl_cql_main.py>`__ 
+
+BENCHMARK ALGORITHM PERFORMANCE
+===============
+
+-  Walker2d
+
+   - walker2d-medium-expert-v0 + CQL
+
+   .. image:: images/walker2d_medium_expert_cql.png
```
Comment (karroyan) [inline D4RL (Mujoco).md]: The image is not shown properly

File: D4RL (Mujoco).rst [hunk 13]
```diff
@@ -0,0 +1,343 @@
+D4RL (Mujoco)
+~~~~~~~~~~~~~~~
```
Comment (karroyan) [inline D4RL (Mujoco).rst:2]: The file should be moved to path: DI-engine-docs\source\13_envs

File: D4RL (Mujoco).rst [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+
+Abstract
+==========
+D4RL is an open source benchmark for offline Reinforcement Learning, which provides a standardized environment and dataset for training and benchmarking algorithms. The collecting strategy of datasets includes:
+
+1. being generated by hand-designed rules and expert demonstrations
+2. agents performing different tasks in the same environment
+3. using hybrid strategies
```
Comment (karroyan) [inline D4RL (Mujoco).rst:10]: "1. datasets generated by ...
2. multi-task datasets( agents performing different...
3. datasets collected with hybrid strategies"

File: D4RL (Mujoco).rst [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+
+-  Maze2D
+-  AntMaze
+-  Gym-MuJoco
+-  Adroit
+-  FrankaKitchen
+-  Flow
+-  Offline CARLA
+
+Note: offline RL is the dataset with d4rl for training, and tests are interacted with a specific RL environment, such as Mujoco.
```
Comment (karroyan) [inline D4RL (Mujoco).rst:22]: "offline RL is trained on the d4rl dataset, and tested by interacting with a specific ..."

File: D4RL (Mujoco).rst [hunk 16]
```diff
@@ -0,0 +1,343 @@
...
+
+-  expert: train a \ `SAC <https://arxiv.org//abs/1801.01290>`__\  algorithm online until the strategy reaches the expert performance level, using the expert strategy to collect 1 million samples of data
+-  medium-expert: mix equal amounts of data collected by expert and medium strategies
+-  medium: first train a SAC algorithm online, stop training in the middle, and then use this partially trained strategy to collect 1 million samples of data
+-  medium-replay：train a SAC algorithm online until the strategy reaches a moderate performance level, collecting all the samples placed in the buffer during training
+-  random：use a random initialization strategy to collect
+
+The picture below shows one of the Hopper games.
+
+.. image:: ./images/d4rl.gif
```
Comment (karroyan) [inline D4RL (Mujoco).rst:35]: The picture has not shown properly

File: D4RL (Mujoco).rst [hunk 17]
```diff
@@ -0,0 +1,343 @@
...
+    # Each task is associated with a dataset
+    # dataset contains observations, actions, rewards, terminals, and infos
+    dataset = env.get_dataset()
+    print(dataset['observations']) # An N x dim_observation Numpy array of observations
+
+    # Alternatively, use d4rl.qlearning_dataset which
+    # also adds next_observations.
+    dataset = d4rl.qlearning_dataset(env)
+
+Mirroring
```
Comment (karroyan) [inline D4RL (Mujoco).rst:135]: Images

File: D4RL (Mujoco).rst [hunk 18]
```diff
@@ -0,0 +1,343 @@
...
+-------------
+
+The DI-engine is ready with images of the framework itself, which can be obtained by \ ``docker pull opendilab/ding:nightly-mujoco``, or by visiting \ `docker
```
Comment (karroyan) [inline D4RL (Mujoco).rst:138]: 1. mirror->image
2. add space after the url

File: D4RL (Mujoco).rst [hunk 19]
```diff
@@ -0,0 +1,343 @@
...
+hub <https://hub.docker.com/r/opendilab/ding>`__\ for more mirrors
+
+
+GYM-MUJOCO'S SPACE BEFORE TRANSFORMATION (ORIGINAL ENVIRONMENT)
```
Comment (karroyan) [inline D4RL (Mujoco).rst:142]: The letters in titles do not need to be fully capitalized

File: D4RL (Mujoco).rst [hunk 20]
```diff
@@ -0,0 +1,343 @@
...
+---------------------
+
+-  The vector consists of physical information(3D position, orientation, and joint angles etc.), and the specific size is \ ``(N, )``\, which is determined by the environment. The data type is \ ``float64``
+-  `Fujimoto <https://github.com/opendilab/DI-engine/blob/main/dizoo/d4rl/entry/d4rl_cql_main.py>`__ mentions that doing obs norm for the d4rl dataset will improve the stability of offline training.
+
+
+Action Space
+----------------
+
+-  The vector consists of physical information(torque etc.), which is often the continuous action space.（N varies with specific subenvironments), the data type is \ ``float32``\ ，需要传入 np 数组（例如动作为\ ``array([-0.9266078 , -0.4958926 ,  0.46242517], dtype=float32)``\ ）
```
Comment (karroyan) [inline D4RL (Mujoco).rst:156]: 1. "..., which is often a continuous action space with N dimension (N varies...."
2. translate the chinese here

File: D4RL (Mujoco).rst [hunk 21]
```diff
@@ -0,0 +1,343 @@
...
+
+-  If it's in the Hooper, N is 3, then the action value be chosen in \ ``[-1, 1]``\.
```
Comment (karroyan) [inline D4RL (Mujoco).rst:158]: "For example, if ..."

File: D4RL (Mujoco).rst [hunk 22]
```diff
@@ -0,0 +1,343 @@
...
+
+Others
+-----
+
+-  The end of the game is the end of the current environment episode
+
+KEY FACTS
+============
+
+1. Vector physical information input, empirically it is not advisable to subtract the mean value in doing norm
```
Comment (karroyan) [inline D4RL (Mujoco).rst:173]: doing norm->normalization

File: D4RL (Mujoco).rst [hunk 23]
```diff
@@ -0,0 +1,343 @@
...
+Others
+---------
+
+-  The \ ``info``\ returned by the environment \ ``step``\ method must contain the \ ``final_eval_reward``\ key-value pair, which represents the evaluation metrics for the entire episode, which in Mujoco is the cumulative sum of the rewards for the entire episode.
+
+
+OTHERS
+========
+
+Inert initialization
```
Comment (karroyan) [inline D4RL (Mujoco).rst:225]: Lazy initialization

File: D4RL (Mujoco).rst [hunk 24]
```diff
@@ -0,0 +1,343 @@
...
+----------------------
+
+To facilitate support for parallel operations such as environment vectorization, environment instances are generally implemented with inert initialization, i.e., the \``__init__`` method does not initialize the real original environment instance, but only sets the relevant parameters and configuration values, and initializes the concrete original environment instance when the \``reset`` method is called for the first time.
```
Comment (karroyan) [inline D4RL (Mujoco).rst:228]: 1. inert initialization -> lazy initialization
2. all the format like __init__ is not shown properly

File: D4RL (Mujoco).rst [hunk 25]
```diff
@@ -0,0 +1,343 @@
...
+
+
+Video storage
+----------------
+
+After the environment is created and before it is reset, the \``enable_save_replay`` method is called to specify the path where the game footage is saved. The environment will automatically save the session's video files after each episode. (The default call \``gym.wrappers.RecordVideo`` is implemented), the code shown below will run an environment episode and save the results of that episode in \``. /video/`` \\.
+.. code:: python
```
Comment (karroyan) [inline D4RL (Mujoco).rst:235]: The code format is not shown properly

File: D4RL (Mujoco).rst [hunk 26]
```diff
@@ -0,0 +1,343 @@
...
+`link <https://github.com/opendilab/DI-engine/blob/main/dizoo/d4rl/entry/d4rl_cql_main.py>`__ 
+
+BENCHMARK ALGORITHM PERFORMANCE
+=================================
+
+-  Walker2d
+
+   - walker2d-medium-expert-v0 + CQL
+
+   .. image:: images/walker2d_medium_expert_cql.png
```
Comment (karroyan) [inline D4RL (Mujoco).rst:339]: The image is not shown properly

**Comment by norman26625** [comment]
rename file (file name should be consistent with other files in the directory )

**Comment by norman26625** [comment]
add banner in the index of the peer directory

**Comment by norman26625** [comment]
check whether the title exists in the idnex of the same level directory

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**